zokrates_core/src/lib.rs

```diff
@@ -1,4 +1,4 @@
-#![feature(box_patterns, box_syntax)]
+#![feature(box_patterns, box_syntax, associated_type_bounds)]
```

Comment:
Maybe there's a way to do without this. It's needed for the `Flatten` trait in `flatten.rs` so that we can have the bound `TryFrom<TypedExpression<'ast, T>, Error: std::fmt::Debug>`

Reply:
At this point there is no real need to have the bound generic right?

Shouldn't this also work:
```
rait Flatten<'ast, T: Field>
    : TryFrom<TypedExpression<'ast, T>, Error=()> 
    + IfElse<'ast, T> 
    + Select<'ast, T> {
```
And then there is no need for `associated_type_bounds`

Reply:
Oh right, we do know the associated type here! Thanks :)

Downside was an experimental feature @JacobEberhardt